sphinx/builders/htmlhelp.py

```diff
@@ -212,7 +212,10 @@ def open_file(self, outdir, basename, mode='w'):
 
     def update_page_context(self, pagename, templatename, ctx, event_arg):
         # type: (unicode, unicode, Dict, unicode) -> None
-        ctx['encoding'] = self.encoding
+        if self.config.htmlhelp_ascii_output:
+            ctx['encoding'] = 'ascii'
+        else:
+            ctx['encoding'] = self.encoding
```

Comment:
If my understanding is right, this is a hack to choose encoding via configuration. If true, I feel `htmlhelp_encoding` is much better for this case.